code/modules/client/preferences.dm

```diff
@@ -547,6 +547,9 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 				else
 					HTML += "<font color=red>[rank]</font></td><td><font color=red><b> \[NON-HUMAN\]</b></font></td></tr>"
 				continue
+			if((rank == "AI") && !(config.allow_ai))
+				HTML += "<font color=red>[rank]</font></td><td><font color=red><b> \[DISABLED\]</b></font></td></tr>"
+				continue

```

Comment:
but the same basic idea is used for all the other things here -- BANNED, NON-HUMAN, etc so it would not solve the exploit problem